libnmstate/nm/device.py

```diff
@@ -168,57 +168,49 @@ def _activate_async(dev):
 
 def delete_device(nmdev):
     mainloop = nmclient.mainloop()
-    mainloop.push_action(_safe_delete_device_async, nmdev)
+    ifname = nmdev.get_iface()
+    if ifname:
```

Comment:
If the request to delete the device is ignored, this should be logged. However, IMHO there are too many checks about whether the device needs to be deleted. IMHO the caller should decide whether this needs to be called at all (which would mean to take the decision making into the main loop in the callback of the profile deactivation IIRC).